src/docker.rs

```diff
@@ -130,31 +143,18 @@ pub fn run(target: &Target,
         docker.args(&opts);
     }
 
-    let mut runner = None;
-
-    if let Some(toml) = toml {
-        for var in toml.env_passthrough(target)? {
-            if var.contains("=") {
-                bail!("environment variable names must not contain the '=' character");
-            }
-
-            if var == "CROSS_RUNNER" {
-                bail!("CROSS_RUNNER environment variable name is reserved and cannot be pass through");
-            }
-
-            // Only specifying the environment variable name in the "-e"
-            // flag forwards the value from the parent shell
-            docker.args(&["-e", var]);
-        }
+    docker
+        .args(&["-e", &format!("CROSS_RUNNER={}", runner.unwrap_or_else(|| String::new()))])
+        .args(&["-v", &format!("{}:/xargo:Z", xargo_dir.display())])
+        .args(&["-v", &format!("{}:/cargo:Z", cargo_dir.display())]);
 
-        runner = toml.runner(target)?;
+    // Prevent `bin` from being mounted inside the Docker container. This is
+    // not required for Podman.
```

Comment:
Why isn't this required?

Reply:
I think it's less that it wasn't required and more that if it's there, podman will fail. This was discovered in #298 where with this setting, podman would fail with:

`Error: error checking path "/cargo/bin": stat /cargo/bin: no such file or directory`

The user found the work-around of not specifying it. I'll acknowledge that I'm not certain if there are any consequences to not adding it.

Reply:
This is specifically so that this directory isn't mounted on macOS, since the container would then try running macOS binaries.

Reply:
Thanks for the clarification. Would an acceptable solution be to only add /cargo/bin to the volumes if /cargo/bin already exists on the host?

Reply:
The problem is *that* it already exists on the host. Maybe we could glob the directory and mount all subdirectories explicitly except `bin`.

Reply:
Oh right, then we have the problem that we can't mount files like `config`.

Reply:
Does `-v /dev/null:/cargo/bin` work?

Reply:
Yep, that seems to do the trick. Should we make this change for both Docker and Podman, or do we want to use the old command for Docker and this new one for Podman?

Reply:
Let's use the same for both.

Reply:
Scratch that. Mounting to /dev/null worked on the CLI, but in `cross` failed with it.

```
Error: cannot configure rootless cgroup using the cgroupfs manager
creating file '/home/ostrosco/.local/share/containers/storage/overlay/34c319fde31eb6cdef406999cddcbcc0fc8886d6b4a9c93a05a3c0f56d6d4936/merged/cargo/bin': Is a directory: OCI runtime error
```

I assume this is because /dev/null isn't a directory. I'm not sure if /var/empty is a better choice?

Reply:
Also I did find that the lack of support for anonymous volumes was a bug in podman that was fixed in the latest release a few days ago: https://github.com/containers/libpod/issues/4276

Not sure if the best approach is to only support this version?

Reply:
If `-v /cargo/bin` works with the latest `podman`, we should require that version instead of the hacky `/dev/null` workaround.

Reply:
Sounds good. Just finished testing on Podman 1.6.3 and everything's working as expected. I just pushed an update to always add /cargo/bin.